### PR TITLE
Replace usages of emberPrintln with ChipLogProgress calls

### PR DESCRIPTION
**Pull request by andy31415**
CHIP debug logs should use ChipLogProgress/Detail/Error rather than ember macros. Only few usages of ember macros
still existed.

Changes:
  - remove ember-print.cpp/h 
  - replace emberAf*PrintLn with ChipLogProgress - this is what the code was doing in the end  and category was ignored
  - emberAfPrintAttributeTable was defined but never used. Removing it.

previous per-cluster code seems to have been guarded by `EMBER_AF_PRINT_ENABLE` which was not set, hence the increase in size of the lock example. However it seemed odd to actually disable printouts of several error messages because of this, so for now I will assume that the print disabling was an oversight (especially for errors). We can iterate and convert things to LogDetail or guard logs with additional defines as a followup.

### CODE REVIEW

**Comment by andy31415** [comment]
> Moving to draft: these macros run VERY deep:
> 
> ```shell
> rg 'ember.*Println' | sd '.*(ember.*Println).*' '$1' | sort | uniq | wc -l
> 124
> ```

Actually not that bad:

```sh
rg 'ember.*Println' src examples | sd '.*(ember.*Println).*' '$1' | sort | uniq | wc -l
8
```

ther usage seems to only be in the gecko sdk. Will try replacing them.

**Review by jmartinez-silabs — APPROVED** [review]
Nit: Do we want to "promote" some of them to ChipLogError

**Comment by andy31415** [comment]
> Nit: Do we want to "promote" some of them to ChipLogError

I believe we may, however maybe as a followup by people with domain knowledge.
I did a blind replace of what was existing before and AfPrint was routed to `ChipLogProgress(Zcl` hence my change.
I expect we eventually want to update some to either detail (if they are noisy) or error (if they are actually errors).

File: src/app/clusters/mode-select-server/mode-select-server.cpp [hunk 1]
```diff
@@ -155,7 +155,7 @@ void emberAfModeSelectClusterServerInitCallback(EndpointId endpointId)
                 {
                     if (onOffValueForStartUp && !onMode.IsNull())
                     {
-                        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "ModeSelect: CurrentMode is overwritten by OnMode");
+                        ChipLogProgress(Zcl, "ModeSelect: CurrentMode is overwritten by OnMode");
```
Comment (tcarmelveilleux) [inline src/app/clusters/mode-select-server/mode-select-server.cpp:158]: _DEBUG should be ChipLogDetail everywhere.

**Review by tcarmelveilleux — APPROVED** [review]

File: src/app/clusters/mode-select-server/mode-select-server.cpp (hunk 1 above)
Comment (andy31415) [reply src/app/clusters/mode-select-server/mode-select-server.cpp:158]: That is what I tought at first, however println seemed to have defaulted to ChipLogProgress and ignored the `category` which is DEBUG in this case, so I redirected everything to progress.

will double-check.

Comment (andy31415) [reply src/app/clusters/mode-select-server/mode-select-server.cpp:158]: So code is:

```
void emberAfPrint(int category, const char * format, ...)
{
    if (format != nullptr)
    {
        va_list args;
        va_start(args, format);
        chip::Logging::LogV(chip::Logging::kLogModule_Zcl, chip::Logging::kLogCategory_Progress, format, args);
        va_end(args);
    }
}
```

where category seems completely ignored and everything goes to ZCL/Progress.

Comment (andy31415) [reply src/app/clusters/mode-select-server/mode-select-server.cpp:158]: Updated PR description.

lock app increase was due to `EMBER_AF_PRINT_ENABLE` being actually disabled and no prints being made.

File: src/app/clusters/door-lock-server/door-lock-server.cpp [hunk 2]
```diff
@@ -346,11 +346,11 @@ void DoorLockServer::setUserCommandHandler(chip::app::CommandHandler * commandOb
                                            const chip::app::Clusters::DoorLock::Commands::SetUser::DecodableType & commandData)
 {
     auto & userIndex = commandData.userIndex;
-    emberAfDoorLockClusterPrintln("[SetUser] Incoming command [endpointId=%d,userIndex=%d]", commandPath.mEndpointId, userIndex);
+    ChipLogProgress(Zcl, "[SetUser] Incoming command [endpointId=%d,userIndex=%d]", commandPath.mEndpointId, userIndex);
 
     if (!SupportsUSR(commandPath.mEndpointId))
     {
-        emberAfDoorLockClusterPrintln("[SetUser] User management is not supported [endpointId=%d]", commandPath.mEndpointId);
+        ChipLogProgress(Zcl, "[SetUser] User management is not supported [endpointId=%d]", commandPath.mEndpointId);
```
Comment (bzbarsky-apple) [inline src/app/clusters/door-lock-server/door-lock-server.cpp:353]: Wondering whether this and some of the other error cases here should be ChipLogError.  Followup for that, though.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by yufengwangca — APPROVED** [review]

File: src/app/clusters/door-lock-server/door-lock-server.cpp (hunk 2 above)
Comment (andy31415) [reply src/app/clusters/door-lock-server/door-lock-server.cpp:353]: Yes, a lot of errors were just passed into Println (which went nowhere it seems, but even once enabled they would do progress and that is wrong). We will have to followup if we want to move detail and error.

**Pull request merged**